subprojects/core/src/main/java/org/gradle/api/internal/project/taskfactory/DefaultTaskClassInfoStore.java

```diff
@@ -35,23 +35,21 @@
 import java.util.HashMap;
 import java.util.Map;
 import java.util.Optional;
-import java.util.function.Function;
 
 @NonNullApi
 public class DefaultTaskClassInfoStore implements TaskClassInfoStore {
     private final CrossBuildInMemoryCache<Class<?>, TaskClassInfo> classInfos;
-    private final Function<Class<?>, TaskClassInfo> taskClassInfoFactory = aClass -> createTaskClassInfo(aClass.asSubclass(Task.class));
 
     public DefaultTaskClassInfoStore(CrossBuildInMemoryCacheFactory cacheFactory) {
         this.classInfos = cacheFactory.newClassCache();
     }
 
     @Override
     public TaskClassInfo getTaskClassInfo(Class<? extends Task> type) {
-        return classInfos.get(type, taskClassInfoFactory);
+        return classInfos.get(type, DefaultTaskClassInfoStore::taskClassInfoFactory);
```

Comment:
💅 If we're killing the contstant, can we inline lambda here instead of extracting its body into a static method? An extra indirection doesn't add much clarity value, IMO.